src/item.cpp

```diff
@@ -10879,6 +10889,23 @@ int item::ammo_consume( int qty, const tripoint &pos, Character *carrier )
         return 0;
     }
     const int wanted_qty = qty;
+    const bool is_tool_with_carrier = carrier != nullptr && type->tool;
+
+    if( is_tool_with_carrier && has_flag( flag_USES_NEARBY_AMMO ) ) {
+        const ammotype ammo = ammo_type();
+        if( !ammo.is_null() ) {
+            inventory carrier_inventory = carrier->crafting_inventory();
```

Comment:
Why are you copying the inventory here?
```suggestion
            const inventory &carrier_inventory = carrier->crafting_inventory();
```